Documentation/ApiOverview/Environment/ApplicationContext.rst

```diff
@@ -0,0 +1,383 @@
+first slash is used to instantiate a new `ApplicationContext`. The so-called
+parent `ApplicationContext`. Here you can see how the context is nested:
+
+*   ApplicationContext: `Development/Local/Ddev/Dev2`
+
+    *   Root ApplicationContext: `Development`
+    *   Parent ApplicationContext: `Local/Ddev/Dev2`
+
+        *   Root ApplicationContext: `Local`
+        *   Parent ApplicationContext: `Ddev/Dev2`
+
+            *   Root ApplicationContext: `Ddev`
+            *   Parent ApplicationContext: `Dev2`
+
+As written above the root `ApplicationContext` must always be one of the 3
+values: `Production`, `Development` or `Testing`. With the 2nd nesting at the
+latest, the root `ApplicationContext` here is now `Local` and with the 3rd
+nesting `Ddev`. There is no way to query this `root ApplicationContext` in the
+PHP class :php:`ApplicationContext`! You only have the option of using
+`getParent()` to access the next parent `ApplicationContext` and using
+:php:`(string)getParent()` to return the complete `ApplicationContext` as a
+string. This means that `Local/Ddev/Dev2` is returned at level 2
+and `Ddev/Dev2` at level 3.
+
+..  _read-application-context:
+
+Reading the ApplicationContext
+==============================
+
+TYPO3 itself already queries the `ApplicationContext` in various places, but
+you can also react to the `ApplicationContext` in various places.
+
+..  _read-application-context-php:
+
+PHP
+---
+
+Here are a few examples of how to access the `ApplicationContext` with the
+:php:`Environment` class:
+
+..  code-block:: php
+
+    <?php
+    use TYPO3\CMS\Core\Core\Environment;
+    class SunnyProducts
+    {
+        public function getDiscount(): int
+        {
+            if (Environment::getContext()->isDevelopment()) {
+                return 20;
+            }
+
+            return 4;
+        }
+    }
+
+..  _read-application-context-site-configuration:
+
+Site Config
+-----------
+
+In the "baseVariants" section of the Site Config, the :yaml:`condition`
+property, which specifies under what circumstances a variant applies, is
+exclusively available for the :yaml:`baseVariant` configuration. For example,
+in the provided code, the baseVariant hosted at 'https://dev-1.example.com/'
+is used when the 'applicationContext == "Development/Dev1"' condition is
+fulfilled.
+
+..  code-block:: php
+
+    baseVariants:
+    -
+      base: 'https://dev-1.example.com/'
+      condition: 'applicationContext == "Development/Dev1"'
+
+..  _read-application-context-typoscript:
+
+TypoScript
+----------
+
+..  code-block:: typoscript
+
+    if {
+       value.data = applicationcontext
+       equals = Development/Dev1
+    }
+
+..  code-block:: typoscript
+
+    [applicationContext == "Development/Dev1"]
+    page.10.wrap = <div style="border: 3px red solid;">|</div>
+    [END]
+
+..  _application-context-presets:
+
+Presets
+=======
+
+As mentioned at the beginning, the `ApplicationContext` affects certain TYPO3
```

Comment:
You lost me, what are presets? Maybe explain?

Reply:
Install tool: Configuration Presets

Reply:
I have updated the header